lib/ansible/plugins/filter/combine.yml

```diff
@@ -1,13 +1,14 @@
 DOCUMENTATION:
   name: combine
   version_added: "2.0"
-  short_description: combine two dictionaries
+  short_description: combine two or more dictionaries
   description:
     - Create a dictionary (hash/associative array) as a result of merging existing dictionaries.
+    - Dictionaries to the right (new) have precedence over dictionaries to the left (old).
   positional: _input, _dicts
   options:
     _input:
-      description: First dictionary to combine.
+      description: First dictionary to combine or a list of dictionaries to combine.
```

Comment:
@sschmittsva AIUI `_input` is always a single parameter, in which the function receives everything that's passed into the filter (i.e. before `| combine(...)`), so it only takes a single dict

Reply:
That's what the former documentation wants you to think :) But it allows to receive a list of dicts, cf. https://github.com/ansible/ansible/blob/afd45aca6ada1dd21fc34a9ccb206ba1e185c883/lib/ansible/plugins/filter/core.py#L343 and the example `list_combine_ab_bc_cd`.

Reply:
it was never intended for the input to be a list of dictionaries, for a case like this i would use map/select/reject filters on top. I'm of a mind to 'fix' combine to follow the normal filter semantics.

Reply:
That 'fix' would push more work onto users. For example, you can currently do
```yaml
# Example 1
[defaults_dict] | product(list_of_dicts) | map("combine")
```
With your suggested 'fix', users would need to do
```yaml
# Example 2
list_of_dicts | map("combine", defaults_dict, reverse=true)
```
except that won't work because `combine` lacks the `reverse` parameter to cause the dicts to be combined right-to-left instead of the heretofore left-to-right order. (In fact, I stumbled across #82329 while researching issues in prep for proposing just such an RFE.)

With such a fix in place, I don't see how one would accomplish what "Example 1" does. Would it not be better to 'fix' _other filters_ that can't currently work this way?

Reply:
its 1 to 100 and also many filters are implemented in other projects, I'm not going to jinja2 and every collection to modify them, its always simpler to fix the outlier.

Reply:
Also fair. I may have misunderstood how the 'fix' you're talking about would percolate up to users. I do want to "get it" though, particularly because I maintain some internal filters for our group, and I'd like to do things the Right Way™.

Thanks for your perspective. This discussion would be better held in the [forum](https://forum.ansible.com/). See you there.